genetic_algorithms/algorithms/simulatedannealing.py

```diff
@@ -0,0 +1,53 @@
+import math
+from genetic_algorithms.models.algorithm import Algorithm
+from genetic_algorithms.problems.base.state import State
+from genetic_algorithms.problems.base.model import Model
+from genetic_algorithms.helpers import History
+from typing import Tuple
+import random
+
+
+HISTORY_SIZE = 5
+
+
+class SimulatedAnnealing(Algorithm):
+    def __init__(self):
+        super(SimulatedAnnealing, self).__init__()
+        self.history = History[int](HISTORY_SIZE)
+        # how to set temperature ???
+        self.temperature = 100

```

Comment:
The temperature is a parameter that should be settable according to the problem requirements. It should be injected via a constructor or otherwise. The algorithm itself shouldn't care.